### PR TITLE
4497: How to Apply Wizard Tracking

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As an education benefits SME, I want to be able to understand how users are engaging with form wizards -- specifically here the how to apply wizard.

[Originating Issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/4497)

## Testing done
Manual testing of values being recorded passes locally.

## Screenshots
N/A

## Acceptance criteria
- [x]  Verify toggling of the radio buttons found within the first question "Are you applying for a new benefit or updating your current education benefits?" fires a DL push and event tag with each change
- [x] Verify answers to other radio buttons for subsequent questions are tracked on form completion (form completion = Apply Now click)
- [x] Verify a DL push, and event tag is fired on clicks on links contained within the STEM Scholarship instructions when shown
- [x] Verify null is passed into dataLayer variables for questions that are not answered
 

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: src/applications/edu-benefits/tests/components/EducationWizard.unit.spec.jsx [hunk 1]
```diff
@@ -124,4 +124,40 @@ describe('<EducationWizard>', () => {
...
     expect(tree.subTree('.usa-alert-warning')).not.be.be.false;
   });
+  it('should record user events for newBenefit', () => {
+    const tree = SkinDeep.shallowRender(<EducationWizard />);
+    expect(global.window.dataLayer.length).to.equal(0);
+    answerQuestion(tree, 'newBenefit', 'yes');
+    expect(global.window.dataLayer.length).to.equal(1);
+    answerQuestion(tree, 'newBenefit', 'no');
+    expect(global.window.dataLayer.length).to.equal(2);
+  });
```
Comment (bkjohnson) [inline src/applications/edu-benefits/tests/components/EducationWizard.unit.spec.jsx]: I'm not sure that these tests are clear, or that they're testing what they say that they are testing. Did you try using a stub or a spy for `recordEvent` to make sure that it was called with the right arguments?

**Review by bkjohnson — COMMENTED** [review]
I'm not convinced that having a bunch of different tests that all rely on checking the length of the `dataLayer` is reliable.

File: src/applications/edu-benefits/tests/components/EducationWizard.unit.spec.jsx (hunk 1 above)
Comment (elder-joseph) [reply src/applications/edu-benefits/tests/components/EducationWizard.unit.spec.jsx]: Hey @bkjohnson, that is a fair point. I will add in some tests to make sure that the correct values are present vs just counting the length on each event.

**Review by cvalarida — APPROVED** [review]
Had a quick refactor suggestion, but it looks good to me.

File: src/applications/edu-benefits/components/EducationWizard.jsx [hunk 2]
```diff
@@ -65,6 +74,65 @@ export default class EducationWizard extends React.Component {
...
+      eduFormChange = 'new';
+    } else if (input === 'no') {
+      eduFormChange = 'update';
+    } else if (input === 'extend') {
+      eduFormChange = 'stem-scholarship';
+    } else {
+      return null;
+    }
+    return eduFormChange;
+  };
```
[the remark this answers is not in the document]
Comment (elder-joseph) [reply src/applications/edu-benefits/components/EducationWizard.jsx]: Agreed that this is a cleaner pattern I updated the three affected functions here.

**Review by sonntag-adam — APPROVED** [review]

**Pull request merged**